### PR TITLE
Use spew for more informative debug output

### PR DESCRIPTION
**Pull request by lavalamp**
For @thockin

### CODE REVIEW

File: test/e2e/util.go [hunk 1]
```diff
@@ -242,7 +243,7 @@ func waitForPodRunningInNamespace(c *client.Client, podName string, namespace st
 			return true, nil
 		}
 		if pod.Status.Phase == api.PodFailed {
-			return true, fmt.Errorf("Giving up; pod went into failed status: \n%#v", pod.Status)
+			return true, fmt.Errorf("Giving up; pod went into failed status: \n%s", spew.Sprintf("%#v", pod))
```
Comment (bprashanth) [inline test/e2e/util.go]: optional nit: save the pod from a previous poll and invoked one of the ObjectDiff functions in util/diff

Comment (lavalamp) [inline test/e2e/util.go]: What would that tell you? I'm actually interested in why it's in the failed state.

Comment (bprashanth) [inline test/e2e/util.go]: I'm assuming it was in a non-failed state and flipped to failed, and more than one field of the pod status changed reflecting why it went to failed. Direct pod spew is sometimes hard to parse, especially with multiple containers.

Comment (lavalamp) [inline test/e2e/util.go]: Fair enough, but I think in this case I actually want to slog through all that stuff, in case the diff obscures a detail I need.

Comment (bprashanth) [inline test/e2e/util.go]: Sure, was just anecdotal from my own debugging escapades :)
